components/core/src/clp_s/ReaderUtils.cpp

```diff
@@ -142,22 +150,67 @@ std::shared_ptr<ReaderUtils::SchemaMap> ReaderUtils::read_schemas(std::string co
     return schemas_pointer;
 }
 
-std::vector<std::string> ReaderUtils::get_archives(std::string const& archives_dir) {
-    std::vector<std::string> archive_paths;
-
-    if (false == boost::filesystem::is_directory(archives_dir)) {
-        throw OperationFailed(ErrorCodeBadParam, __FILENAME__, __LINE__);
+namespace {
+std::shared_ptr<clp::ReaderInterface> try_create_file_reader(std::string_view const file_path) {
+    try {
+        return std::make_shared<clp::FileReader>(std::string{file_path});
+    } catch (clp::FileReader::OperationFailed const& e) {
+        SPDLOG_ERROR("Failed to open file for reading - {} - {}", file_path, e.what());
+        return nullptr;
     }
+}
 
-    boost::filesystem::directory_iterator iter(archives_dir);
-    boost::filesystem::directory_iterator end;
-    for (; iter != end; ++iter) {
-        if (boost::filesystem::is_directory(iter->path())) {
-            archive_paths.push_back(iter->path().string());
+bool try_sign_url(std::string& url) {
+    clp::aws::AwsAuthenticationSigner signer{
+            std::getenv(cAwsAccessKeyIdEnvVar),
```

Comment:
noticed that if I forget to pass in the access key,  clp-s will throw a quite confusing message
`terminate called after throwing an instance of 'std::logic_error'
  what():  basic_string::_M_construct null not valid
Aborted`

We should let signer parse the env vars, verify they are not null and then initilaize the member variables